worker/rhpv3.go

```diff
@@ -545,10 +541,11 @@ func (a *accounts) deriveAccountKey(hostKey types.PublicKey) types.PrivateKey {
 }
 
 // priceTable fetches a price table from the host. If a revision is provided, it
-// will be used to pay for the price table. The returned price table is
-// guaranteed to be safe to use.
-func (h *host) priceTable(ctx context.Context, rev *types.FileContractRevision) (rhpv3.HostPriceTable, error) {
-	pt, err := h.priceTables.fetch(ctx, h.hk, rev)
+// will be used to pay for the price table. If not it will be paid for using an
+// ephemeral account, in which case the given amount parameter will get updated.
+// The returned price table is guaranteed to be safe to use.
+func (h *host) priceTable(ctx context.Context, rev *types.FileContractRevision, amount *types.Currency) (rhpv3.HostPriceTable, error) {
```

Comment:
Two things are a bit odd to me about this. First the `amount` being returned by reference when usually we return the cost as a return variable. The other thing is that we don't populate the cost when we pay by contract. Considering that we should also record price table updates in the contract spending it probably makes sense to always return the cost.

Reply:
I updated the return variables and always return the cost. We currently don't have pt updates in the contract spending but it's ready to be used now whenever we add that.